common/proto/src/main/java/org/thingsboard/server/common/adaptor/JsonConverter.java

```diff
@@ -466,6 +466,12 @@ private static JsonObject toJson(ProvisionDeviceResponseMsg payload, boolean toG
         return result;
     }
 
+    public static JsonObject toGatewayDeviceDisconnectJson(String deviceName) {
+        JsonObject result = new JsonObject();
+        result.addProperty(DEVICE_PROPERTY, deviceName);
+        return result;
+    }
```

Comment:
In the same way, as for the proto version, I have expected to see a reason code added in the MQTT message.